### PR TITLE
Better macOS instructions

### PR DESCRIPTION
**Pull request by sedited**
This reworks the macOS instructions. @Lederstrumpf , @Git-Jiro please try these out and amend where necessary.

### CODE REVIEW

**Comment by acey1** [comment]
For usage on an actual Mac:
You may want to consider writing a note for people that already have this installed from prior .app usage to NOT use these commands you have here:


 +    echo -e "rpcuser=bitcoinrpc\nrpcpassword=$(xxd -l 16 -p /dev/urandom)" > "/Users/${USER}/Library/Application Support/gridcoinresearch/gridcoinresearch.conf"

 +    chmod 600 "/Users/${USER}/Library/Application Support/gridcoinresearch/gridcoinresearch.conf"

Under Other Commands, the command "./src/gridcoinresearchd -daemon" will not open the file. However the command "open -a gridcoinresearch"  Will open it. 

This actual macOS .app file is supposed to be located in /src/qt folder and has an actual .app ending. Instead the file found here is a unix executable file.  Why?

**Comment by ghost** [comment]
For usage on an actual Mac:

You may want to change in the first line "build Bitcoin" to "build Gridcoin".

You may want to consider writing a note for people that already have this installed from prior .app usage to NOT use these commands you have here:


 +    echo -e "rpcuser=bitcoinrpc\nrpcpassword=$(xxd -l 16 -p /dev/urandom)" > "/Users/${USER}/Library/Application Support/gridcoinresearch/gridcoinresearch.conf"

 +    chmod 600 "/Users/${USER}/Library/Application Support/gridcoinresearch/gridcoinresearch.conf"

The  gridcoinresearch.conf is created automatically when you open the app anyway, so why the commands?

Under Other Commands, the command "./src/gridcoinresearchd -daemon" will not open the file. However the command "open -a gridcoinresearch"  (note spelling difference) Will open it. 
Why is this a unix file with no icon?

**Comment by tomasbrod** [comment]
> The gridcoinresearch.conf is created automatically when you open the app anyway, so why the commands?

These commands are necessary for daemon usage. They initialize rpc password to a random string and set a proper permissions for safety. Yes, the first command will OVERWRITE the config file, so warning should be put there. Either backup the file and merge manually, or replace the `>` by `>>` to append the file instead.

>Under Other Commands, the command "./src/gridcoinresearchd -daemon" will not open the file. However the command "open -a gridcoinresearch" (note spelling difference) will open it.

The example command will (i suppose) run the build in daemon mode where no wallet window will be shown and the daemon will run in the background (hence the "-daemon" and "d" suffix). If you you invoke the `open` command, I assume (don't know mac) that the system will try to open the system-installed version of the program. `./src/gt/gridcoinresearch` will open the GUI.

> Why is this a unix file with no icon?

Unix executable files do not have a icon.

**Comment by ghost** [comment]
Update----

I Tomas,

Please see my reply below.

Regards,
William





I agree these files are needed, but they are created automatically in Mac when one opens the app. Mac people mostly use GUIs, unlike Linux people. Think of your everyday Mac audience here please.

When I tried the ./src/qt/gridcoinresearchd, per your guide (not ./src/qt/gridcoinresearch, as misspelled above?) command on my Mac, the headless daemon would not open.  One daemon in question is located in the /Gridcoin-Research/src  directory and is spelled “gridcoinresearchd". "./src/gridcoinresearchd" will open this file in both headless and GUI modes simultaneously.

The make command also produces a unix exec file in the /Gridcoin-Research/src/qt directory. This unix exec file is spelled “gridcoinresearch” and is the correct location for the Mac App location to be in I believe. (I have been told both ways, so no one really knows then.) When I open this file by clicking on it, BOTH a GUI and headless daemon opens! It’s version is Halford v0.14.99.0-unk-14, the current testnet version. If I use the command "open -a gridcoinresearch”, only the GUI opens as expected but to version Nakamoto v3.6.2.0-g37a5e5b9e-dirty-317. If I open this same file with the -testnet flag command, " open -a gridcoinresearch --args -testnet” a GUI opens as expected, but with wrong testnet version number, Nakamoto v3.6.2.0-g37a5e5b9e-dirty-317. So the unix exec file does not work properly on a real Mac or I need some unknown command to make this work.

Why is this a unix file with no icon?

Sure, but what I mean is, why is auto build placing a unix file with no icon in /Gridcoin-Research/src/qt when it should be placing a Mac.app file? Understand? I see you have fixed the "make deploy”, if it works then we should get a real mac.app that hopefully works correctly in the DMG.

So in conclusion and thanks to @brod, the "./src/qt/gridcoinresearch -printtoconsole -debug=true -testnet" command works. This should put to rest any doubt about which directory the proper executable is in and what the proper command it is to execute it.

Thank you for your help.

~acey
